packages/terra-menu/src/MenuItem.jsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Arrange from 'terra-arrange';
+import CheckIcon from 'terra-icon/lib/icon/IconCheckmark';
+import ChevronIcon from 'terra-icon/lib/icon/IconChevronRight';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './MenuItem.scss';
+
+const cx = classNames.bind(styles);
+
+const KEYCODES = {
+  ENTER: 13,
+  SPACE: 32,
+};
+
+const contextTypes = {
+  isGroupItem: PropTypes.bool,
+  isSelectableMenu: PropTypes.bool,
+};
+
+const propTypes = {
+  /**
+   * Sets the item's text
+   */
+  text: PropTypes.string,
+
+  /**
+   * Indicates if the item is selected.
+   */
+  isSelected: PropTypes.bool,
+
+  /**
+   * Indicates if the item should be selectable
+  */
+  isSelectable: PropTypes.bool,
+
+  /**
+   * List of Menu.Items to display in a submenu when this item is selected.
+   */
+  subMenuItems: PropTypes.arrayOf(PropTypes.element),
+
+  /**
+   * Callback function for when item is clicked
+   */
+  onClick: PropTypes.func,
+};
+
+const defaultProps = {
+  text: '',
+  isSelected: false,
+  isSelectable: undefined,
+  subMenuItems: [],
+};
+
+class MenuItem extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.wrapOnClick = this.wrapOnClick.bind(this);
+    this.wrapOnKeyDown = this.wrapOnKeyDown.bind(this);
+    this.handleSelection = this.handleSelection.bind(this);
+    this.state = { isSelected: props.isSelected && props.isSelectable && !context.isGroupItem };
+  }
+
+  handleSelection(event) {
```

Comment:
Doesn't need to. That check is being done before wrapping onClick and onKeyDown on this line https://github.com/cerner/terra-core/pull/702/files/3e26dbfcb281f12b23fb3bf98ed2bf60a5e1dcac#diff-29f754ea95dc722a29df6d9782f5cca6R107